zebra/zapi_msg.c

```diff
@@ -2253,12 +2253,19 @@ static inline void zread_rule(ZAPI_HANDLER_ARGS)
 			}
 		}
 
-		if (!is_default_prefix(&zpr.rule.filter.src_ip))
-			zpr.rule.filter.filter_bm |= PBR_FILTER_SRC_IP;
+		if (ifindex) {
```

Comment:
We already have 1 test for if (ifindex) right above, why the second?

Reply:
ok.
I realise that BGP is well configuring default IPv4 prefix in src field. so we can avoid modifying the code as it is today.